### PR TITLE
Multiplayer/ ctp 329   interaction system

### PR DESCRIPTION
**Pull request by KyriakosNU**

### CODE REVIEW

File: CSC8503/LevelManager.cpp [hunk 1]
```diff
@@ -276,7 +276,7 @@ void LevelManager::LoadLevel(int levelID, int playerID, bool isMultiplayer) {
 		}
 	}
 
-	LoadGuards((*mLevelList[levelID]).GetGuardCount(), isMultiplayer);
+	//LoadGuards((*mLevelList[levelID]).GetGuardCount(), isMultiplayer);
```
Comment (Nijelous) [inline CSC8503/LevelManager.cpp:279]: Do we still need this commented out?

File: CSC8503/NetworkPlayer.cpp [hunk 2]
```diff
@@ -279,66 +279,69 @@ void NetworkPlayer::RayCastFromPlayer(GameWorld* world, float dt) {
...
+			}
 
-				//Check if object is an item.
-				Item* item = dynamic_cast<Item*>(objectHit);
-				if (item != nullptr) {
-					item->OnPlayerInteract(mPlayerID);
-					return;
-				}
+			//Check if object is an item.
+			Item* item = dynamic_cast<Item*>(objectHit);
```
Comment (Nijelous) [inline CSC8503/NetworkPlayer.cpp:317]: We're avoiding dynamic_cast due to PS5 not liking it, find another way to cast this such as static_cast

File: CSC8503/NetworkPlayer.cpp [hunk 3]
```diff
@@ -279,66 +279,69 @@ void NetworkPlayer::RayCastFromPlayer(GameWorld* world, float dt) {
...
-				Interactable* interactablePtr = dynamic_cast<Interactable*>(objectHit);
-				if (interactablePtr != nullptr && interactablePtr->CanBeInteractedWith(interactType, this)) {
-					interactablePtr->Interact(interactType, this);
-					if (interactType == ItemUse) {
-						mInventoryBuffSystemClassPtr->GetPlayerInventoryPtr()->UseItemInPlayerSlot(mPlayerID, mActiveItemSlot);
-					}
-
-					return;
+			//Check if object is an interactable.
+			Interactable* interactablePtr = dynamic_cast<Interactable*>(objectHit);
```
Comment (Nijelous) [inline CSC8503/NetworkPlayer.cpp:324]: Same here

File: CSC8503/NetworkPlayer.cpp [hunk 4]
```diff
@@ -279,66 +279,69 @@ void NetworkPlayer::RayCastFromPlayer(GameWorld* world, float dt) {
...
-								otherPlayerObject->GetActiveItemSlot(), this->GetPlayerID());
-					}
-				}
+				return;
+			}
 
-				std::cout << "Object hit " << objectHit->GetName() << std::endl;
+			if (interactType == PickPocket)
+			{
+				NetworkPlayer* otherPlayerObject = dynamic_cast<NetworkPlayer*>(objectHit);
```
Comment (Nijelous) [inline CSC8503/NetworkPlayer.cpp:336]: And here

**Review by Nijelous — APPROVED** [review]
Rest looks good just please remove the dynamic casts and check for things being commented out that might not need to be.

**Review by erendgrmnc — APPROVED** [review]

**Pull request closed without merging**